src/main/kotlin/it/pagopa/wallet/services/WalletEventSinksService.kt

```diff
@@ -0,0 +1,60 @@
+package it.pagopa.wallet.services
+
+import it.pagopa.wallet.audit.LoggedAction
+import it.pagopa.wallet.config.properties.RetrySavePolicyConfig
+import it.pagopa.wallet.repositories.LoggingEventRepository
+import java.time.Duration
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.context.event.ApplicationReadyEvent
+import org.springframework.context.ApplicationListener
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.core.publisher.Sinks
+import reactor.core.scheduler.Schedulers
+import reactor.util.retry.Retry
+
+@Service
+class WalletEventSinksService(
+    @Autowired private val loggingEventRepository: LoggingEventRepository,
+    @Autowired private val retrySavePolicyConfig: RetrySavePolicyConfig
+) : ApplicationListener<ApplicationReadyEvent> {
+    val walletEventSink: Sinks.Many<LoggedAction<*>> = Sinks.many().unicast().onBackpressureBuffer()
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    fun <T : Any> tryEmitEvent(loggedAction: LoggedAction<T>): Mono<LoggedAction<T>> =
+        Mono.defer {
+                walletEventSink.tryEmitNext(loggedAction)
```

Comment:
What are we trying to accomplish by applying this refactor?

Reply:
Enhance visibiliti of errors when using tryEmitNext. 

The doOnError block handles logging for exceptions that occur during the emission process, while the check for result.isFailure specifically logs if the tryEmitNext call fails. These are slightly different scenarios:

result.isFailure: This logs if the tryEmitNext method itself fails to emit the event. This failure is specific to the emission attempt and doesn't necessarily throw an exception.
doOnError: This logs any exceptions that occur during the entire Mono.defer block, including but not limited to the emission attempt.
While there is some overlap, having both can provide more granular logging and help diagnose different types of issues. However, if you prefer to avoid redundancy and simplify the logging, you could choose to keep only one of these logging mechanisms based on your specific needs.

Reply:
Refactor done in [814b03c](https://github.com/pagopa/pagopa-wallet-service/pull/195/commits/814b03ccca2baf5c83d195f71ef2b89c9f6d5f98), I also added tests for those specific scenarios. Let me know what you think.